src/Orleans.Runtime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -453,7 +453,17 @@ private async Task<bool> ReadFromQueue(QueueId myQueueId, IQueueAdapterReceiver
                 if (pubSubCache.TryGetValue(streamId, out streamData))
                 {
                     streamData.RefreshActivity(now);
-                    StartInactiveCursors(streamData, startToken); // if this is an existing stream, start any inactive cursors
+                    if (streamData.StreamRegistered)
+                    {
+                        StartInactiveCursors(streamData,
+                            startToken); // if this is an existing stream, start any inactive cursors
+                    }
+                    else
+                    {
+                        this.logger.LogInformation($"Pulled new messages in stream {streamId} from the queue, but pulling agent haven't succeeded in" +
```

Comment:
Should this rather be a warning?

Reply:
yeah it could be. if code hit this, then there might be message delay, but not message loss. So warning should be proper.